ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CommandDeduplicationIT.scala

```diff
@@ -69,23 +70,25 @@ final class CommandDeduplicationIT(
         _.commands.deduplicationPeriod :=
           DeduplicationPeriod.DeduplicationDuration(deduplicationDuration.asProtobuf)
       )
-    val acceptedSubmissionId1 = newSubmissionId()
+    val firstAcceptedSubmissionId = newSubmissionId()
+    val firstSubmissionSendTime = Instant.now()
```

Comment:
Because the record time is not necessarily aligned with the clock that's used for converting durations into offsets.

Reply:
I think we should properly distinguish the different ways to report the deduplication period:
* If the deduplication period requested as a duration and is reported as a duration, then we can just compare the two durations.
* If the deduplication period is requested as an offset and we report a duration, then we can currently also use the record time because only kvuils does this and kvutils uses the record time for the conversion.
* If the deduplication period is requested as an offset and reported as an offset, then just compare the offsets.
* If the deduplication period is requested as a duration and reported as an offset, then we can't use the record times (Canton case) because record times need not increase monotonically in Canton. IIRC we agreed on using the local clock of the test runner (which should be sufficiently synchronized with the participant's local clock used to convert durations into offsets). That's why `Instant.now()` makes sense, even though I'd find `System.nanoTime` better.

Reply:
Thanks, @andreaslochbihler-da. Is it just an explanation or I should change something?

Reply:
This here is just an explanation. I've left more detailed comments below where something should be changed.

Reply:
Just one more question here, @andreaslochbihler-da. Why do you prefer `System.nanoTime`? We usually use `Instant` across the codebase.

Reply:
Just look at the scaladocs for both. `System.nanoTime` is better for measuring how long something takes because it's guaranteed that the underlying clock will not run backwards. `Instant.now` can jump forwards and backwards, e.g., if there are leap seconds or the VM's clock is adjusting. In your test case, you are only interested in elapsed time, not in an actually meaningful moment in time. See https://stackoverflow.com/a/1776053 for a good discussion of the difference.

Reply:
Thanks for the thorough explanation. I would like to keep `Instant`s, as the code is easier and cleaner this way, if you don't mind.

Reply:
Sure. This is your team's code.